File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    words =
      Regex.split(
        ~r/[^[:alnum:]-]/u,
        String.downcase(sentence),
        trim: true
      )

    Enum.reduce(words, %{}, fn word, acc ->
      Map.update(acc, word, 1, &(&1 + 1))
    end)
  end
end

File: javascript/word-count/word-count.js
export const countWords = (words_string) => {
  const words = words_string
    .toLowerCase()
    .match(/\b\w+('\w)?\b/g)

  return words.reduce((acc, word) => {
    return { ...acc, [word]: !!acc[word] ? acc[word] + 1 : 1 }
  }, {});
};
